src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -214,8 +215,27 @@ public override void SetLength(long value)
         public override void Write(byte[] buffer, int offset, int count)
         {
             var pos = InnerStream.Position;
-            InnerStream.Write(buffer, offset, count);
-            _length = Math.Max(_length, pos + count);
+            try
+            {
+                InnerStream.Write(buffer, offset, count);
+                _length = Math.Max(_length, pos + count);
+            }
+            catch (IOException e)
+            {
+                if (IsDiskFull(e))
+                {
+                    throw new OutOfDiskSpaceException("Out of disk space", e);
+                }
+            }
+        }
+
+        private static bool IsDiskFull(Exception ex)
+        {
+            const int HR_ERROR_HANDLE_DISK_FULL = unchecked((int)0x80070027);
```

Comment:
This should be added in `IsOutOfDiskSpaceException` (for `Windows` only)

Reply:
There's already platform check in IsOutOfDiskSpaceException method.